0022-circuit-block.md

```diff
@@ -0,0 +1,300 @@
+> This is absolutely not necessary to finalise for this RFC, but I feel like it's worth thinking about this as a joined-up story, since previously we have treated the output of `transpile` as "ready for QPU execution", but annotations stymie that.
+
+If we go this route, this might motivate a `BackendV3` definition which includes information on what annotations should be permitted to remain.
+This has some benefits:
+
+* the transpiler can reject the output from malformed pass-manager pipelines, or input circuits where the user put on a custom annotation that the backend does not know how to resolve.  This allows client-side validation of the circuits.
+* the QPY API could be extended to retrieve the serialisers/deserialisers from a `backend` object, rather than needing a user to construct it themselves.
+
+If we _truly_ associated the Qiskit object `Backend` with a "quantum computer" (as opposed to `Target`'s QPU), we might even consider an API where the primitives entry points are moved onto the `BackendV3` object, as in:
+
+```python
+class BackendV3:
+    @property
+    @abstractmethod
+    def target(self) -> Target: ...
+    @abstractmethod
+    def allowed_annotations(self) -> AnnotationChecker: ...
+    @abstractmethod
+    def sample(self, sampler_pub_likes) -> SamplerJob: ...
+    @abstractmethod
+    def estimate(self, estimator_pub_likes) -> EstimatorJob: ...
+```
+
+This might naturally extend to such a `BackendV3` providing the additional information that is needed for Qiskit SDK to safely transpile entire sampler and estimator pubs; the compilation of a pub _also_ implies further processing will be done by a quantum computer, just like an annotation remaining in a single circuit.
+
+
+## Case study: twirling, noise learning, and mitigation with `qiskit-ibm-runtime`
+
+We envision two main workflows that users will follow once `Box`es become available, one for regular users and another one for power users:
+
+1. The workflow for regular users:
+    - Users initialize a `QuantumCircuit` without boxes, as they do today.
+    - They apply all of the desired transpiler passes, for example to map the circuit to an ISA circuit for the backend that they wish to use.
+    - They use (a convenience method built around) a new transpiler pass that collects the circuit's gates into boxes, with the ability of specifying different collection strategies.
+    - They submit their job.
+2. The workflow for power users:
+    - Users initialize a `QuantumCircuit` adding boxes manually as they wish.
+    - They submit their job.
+
+Setting up these two workflows presents notable advantages, such as:
+- All of the existing transpiler passes can be asked (at least initially) to simply ignore the boxes, since transpilation is meant to happen before the boxes are generated. This can save quite a lot of time that would otherwise be spent adding logic to every single transpiler pass -- and yet, we can choose to do this in the future.
```

Comment:
This can probably be removed, now that we use control flow ops?

Reply:
it might be worthwhile to instead say `to simply ignore the boxes with twirling annotations` or some verbiage to connect it back to [this line](https://github.com/SamFerracin/RFCs/pull/1/files#diff-8468c13b6726725a9dfdf5968d828e7a52b13e30d77470e9e6f50d126179d56aR254)

Reply:
I didn't touch the text in this section - when I'm back, I can review any changes in here that you want to make in the meantime?